### DISCUSSION TITLE
What is the right place for adding helper class?

### DISCUSSION THREAD

**Discussion by borahh** [TukuToi/better-wp-plugin-boilerplate#27] [Q&A] [answered]

**Comment by smileBeda** [comment, marked as answer]
The way I do it if I need other than the existing public or admin classes, Is that i add a new file with the new class in the respective folder (admin or public) and load the new class in the main includes file where all other classes are loaded.

you could also include it in one of the existing classes (as a dependency).

you can see an example where I did this here 
https://github.com/TukuToi/tukutoi-shortcodes

In this case here I extended the loader class with an autoloader instead:
https://github.com/TukuToi/tukutoi-cp-directory-integration
so you don’t need to worry about inclusion anymore…

**Reply by borahh** [reply]
How do you register custom post types with this boilerplate?

**Reply by smileBeda** [reply]
Using the WordPress API, and putting the add_action into the main includes file (admin section), then the code for it in a function in either the main admin class, or a custom class of yours.

See this example https://github.com/TukuToi/tukutoi-template-builder/blob/7f2b182d5f6783c91564d54dbd8bf6dfb840181a/admin/class-tkt-template-builder-admin.php#L600, it is then called in https://github.com/TukuToi/tukutoi-template-builder/blob/7f2b182d5f6783c91564d54dbd8bf6dfb840181a/includes/class-tkt-template-builder.php#L237

**Reply by borahh** [reply]
Thank you so much!

**Reply by smileBeda** [reply]
You are welcome. 

Please feel free to mark the reply adequate as "Mark as answer", so others seeking the same information can find it quicker.

Thanks!
